### DISCUSSION TITLE
Setting `shm_size` for Docker Containers?

### DISCUSSION THREAD

**Discussion by ckcollab** [community/community#25740] [Actions]
<p>Using Selenium browser to run tests, apparently, takes quite a bit of memory. Without the <code>shm_size</code> flag to increase allowed shared memory I get crashes from within Selenium (out of memory errors).</p>
<p>Is there a way within GitHub Actions to set this value?</p>
<p>Here’s how I do it locally in my <code>docker-compose.yml</code>:</p>
<pre><code class="lang-auto">  selenium:
    image: selenium/standalone-chrome-debug
    shm_size: 2gb
    ports:
      - 4444:4444
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>I assume you mean for a job container? I haven’t tried myself, but based on the docs <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idcontaineroptions"><code>jobs.&lt;job_id&gt;.container.options</code></a> should do the trick:</p>
<pre><code class="lang-auto">container:
  image: selenium/standalone-chrome-debug
  options: --shm-size=2gb
</code></pre>
<p>Service containers have an <code>options</code> parameter that should work the same way.</p>

**Comment by ckcollab** [comment]
<p>So simple, thank you! I was worried for a second :slight_smile:</p>

**Comment by uuf6429** [comment]
What about `docker-compose`?

@airtower-luna I see a lot of issues on GitHub regarding docker that have been answered with "use `jobs.<job_id>.container....`" this is a very bad practice in my opinion, because it ties implementations to the vendor (GitHub) AND because from GitHub side, it is reinventing the wheel (docker-compose).
This isn't just political/vendor lock-in, but it also encourages GitHub users to have a different flow between GitHub actions and testing things locally, which provides a bad developer/collaborator experience.

While writing this, I'm currently searching, (since several days), the reason why running docker-compose in GitHub actions doesn't work for certain services in GitHub but works perfectly fine locally. So in my view, GitHub would have better spent effort on a nicer troubleshooting mechanism instead of reinventing docker-compose. (I'll eventually open a new issue/discussion if I still can't manage)

**Reply by airtower-luna** [reply]
The question was about how to do that in Actions, not Compose, don't blame me for answering the question that was asked. I'd generally prefer to use an identical setup locally and in CI, too. :wink:

**Reply by uuf6429** [reply]
ok, I guess I'm just a bit sensitive to this situation since pretty much any search on the subject ("githab actions docker-compose ...") yields the aforementioned results (to use the vendor-specific option). If it wasn't for some other example actions I saw, I wouldn't even know that you can use docker-compose since github doesn't seem to acknowledge it (in docs).

But it is what it is, I suppose. 🤷

**Reply by uuf6429** [reply]
Although, in my opinion, the answer could have also mentioned how the set up might have been better with docker-compose, especially considering the OP already mentioned they had such a local file at hand - presumably GitHub won't consider 
investing effort in the docker-compose approach (documentation etc) if people do not ask about it more.

To be fair, I could as well open a new topic on the subject instead of hijacking an existing discussion. 😅

**Reply by airtower-luna** [reply]
> If it wasn't for some other example actions I saw, I wouldn't even know that you can use docker-compose since github doesn't seem to acknowledge it (in docs).

It's mentioned in the list of pre-installed tools (the links is for the `ubuntu-22.04` runner): https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#tools
